### PR TITLE
Evolution bindings

### PR DESCRIPTION
**Pull request by hernot**
Ok finally i decided to update my fork on github and create a pull request for the evloulution support stuff as suggested by pschule on mantis bugtracker. (refer to issues 0011008, 0011010, 0011006)

Will do a retest the next weeks if moving to a fork of latest github upstream version didn't break anything in my fixes and that my fixes didn't break anything in that version.

Includes a separate commit for all the changes i had to make to make webpack-server working on my server which is located remotely and can only be accesses through https. Further i have installed a newer verison of webpack server than used by metaways

### CODE REVIEW

**Comment by pschuele** [comment]
hi,
thanks for your pull request. we will test and review this in a couple of days.
cu
Philipp

File: tine20/Tinebase/js/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "uglify-loader": "^1.3.0",
     "unminified-webpack-plugin": "^1.2.0",
     "url-loader": "^0.5.7",
-    "webpack": "^1.13.3",
+    "webpack": "^2.5.1",
```
Comment (cookiekiller) [inline tine20/Tinebase/js/package.json]: no, this doesn't work for most of our usages!

File: tine20/Tinebase/js/webpack.config.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ module.exports = {
         proxy: [
             {
                 context: ['**', '!/webpack-dev-server'],
-                target: 'http://localhost/',
+                target: 'http://tine20debug.out-world.com/',
```
Comment (cookiekiller) [inline tine20/Tinebase/js/webpack.config.js]: please revert this change

File: tine20/Addressbook/Convert/Contact/VCard/Evolution.php [hunk 3]
```diff
@@ -0,0 +1,236 @@
+<?php
+/**
+ * Tine 2.0
+ *
+ * @package     Addressbook
+ * @subpackage  Convert
+ * @license     http://www.gnu.org/licenses/agpl.html AGPL Version 3
+ * @author      Lars Kneschke <l.kneschke@metaways.de>
```
Comment (cookiekiller) [inline tine20/Addressbook/Convert/Contact/VCard/Evolution.php]: feel free to add you as an author here, I guess you aren't lars kneschke, for new files you are free to replace it.

File: tine20/Addressbook/Convert/Contact/VCard/Evolution.php [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+ * @copyright   Copyright (c) 2011-2013 Metaways Infosystems GmbH (http://www.metaways.de)
```
Comment (cookiekiller) [inline tine20/Addressbook/Convert/Contact/VCard/Evolution.php]: 2017

File: tine20/Addressbook/Convert/Contact/VCard/Evolution.php [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+/**
+ * class to convert a Evolution vcard to contact model and back again
+ *
+ * @package     Addressbook
+ * @subpackage  Convert
+ */
+class Addressbook_Convert_Contact_VCard_Evolution extends Addressbook_Convert_Contact_VCard_Abstract
+{
+    // Mozilla/5.0 (X11; Linux i686) KHTML/4.7.3 (like Gecko) Konqueror/4.7
```
Comment (cookiekiller) [inline tine20/Addressbook/Convert/Contact/VCard/Evolution.php]: Please add an example of the evolution user agent

File: tine20/Addressbook/Convert/Contact/VCard/Evolution.php [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+    // KDE DAV groupware client
+    const HEADER_MATCH = '/Evolution\/(?P<version>.*)/';
+    
+    protected $_emptyArray = array(
+        'adr_one_countryname'   => null,# 1 1
```
Comment (cookiekiller) [inline tine20/Addressbook/Convert/Contact/VCard/Evolution.php]: what's up with these comments? any meanings?

File: tine20/Tinebase/Frontend/Http.php [hunk 7]
```diff
@@ -560,8 +560,13 @@ public static function getAssetsMap($asJson=false)
             $jsonFileUri = $devServerURL . '/' . $jsonFile;
             $json = Tinebase_Helper::getFileOrUriContents($jsonFileUri);
             if (! $json) {
-                Tinebase_Core::getLogger()->ERR(__CLASS__ . '::' . __METHOD__ . ' (' . __LINE__ .') Could not get json file: ' . $jsonFile);
```
Comment (cookiekiller) [inline tine20/Tinebase/Frontend/Http.php]: please revert this as well

File: tine20/Tasks/Frontend/WebDAV/Task.php [hunk 8]
```diff
@@ -90,8 +90,11 @@ public static function create(Tinebase_Model_Container $container, $name, $vobje
         }
 
         $task->container_id = $container->getId();
-        $id = ($pos = strpos($name, '.')) === false ? $name : substr($name, 0, $pos);
-        $task->setId($id);
+        $id = $task->getId();
+	if ( $id === null ) {
```
Comment (cookiekiller) [inline tine20/Tasks/Frontend/WebDAV/Task.php]: does the unittest pass?

**Comment by hernot** [comment]
Ok have made the requested changes 

revert the modifications necessary for the setup on my server located remotely
updated the headers, removed the comments i used for verify that i have enabled only those fields supporeted by evolution and tine20.

**Comment by hernot** [comment]
Ok have made the requested changes 

revert the modifications necessary for the setup on my server located remotely
updated the headers, removed the comments i used for verify that i have enabled only those fields supporeted by evolution and tine20.

**Comment by hernot** [comment]
Closed by accident.
